src/metabase/api/collection.clj

```diff
@@ -54,15 +74,15 @@
   [archived namespace]
   {archived  (s/maybe su/BooleanString)
    namespace (s/maybe su/NonBlankString)}
-  (let [archived? (Boolean/parseBoolean archived)]
-    (as-> (db/select Collection
-            {:where    [:and
-                        [:= :archived archived?]
-                        [:= :namespace namespace]
-                        (collection/visible-collection-ids->honeysql-filter-clause
-                         :id
-                         (collection/permissions-set->visible-collection-ids @api/*current-user-permissions-set*))]
-             :order-by [[:%lower.name :asc]]}) collections
+  (let [archived? (Boolean/parseBoolean archived)
```

Comment:
Note that in both cases, a when-guard is in place since the in clause will throw an exception if the collection is empty.